services/src/main/java/org/keycloak/services/resources/admin/AdminConsole.java

```diff
@@ -213,38 +222,49 @@ public Response whoAmI(final @Context HttpHeaders headers) {
             displayName = user.getUsername();
         }
 
-        RealmModel masterRealm = getAdminstrationRealm(realmManager);
-        Map<String, Set<String>> realmAccess = new HashMap<String, Set<String>>();
+        RealmModel masterRealm = new RealmManager(session).getKeycloakAdminstrationRealm();
+
         if (masterRealm == null)
             throw new NotFoundException("No realm found");
-        boolean createRealm = false;
-        if (realm.equals(masterRealm)) {
-            logger.debug("setting up realm access for a master realm user");
-            createRealm = user.hasRole(masterRealm.getRole(AdminRoles.CREATE_REALM));
-            addMasterRealmAccess(user, realmAccess);
-        } else {
-            logger.debug("setting up realm access for a realm user");
-            addRealmAccess(realm, user, realmAccess);
+
+        RealmModel currentRealm = session.realms().getRealmByName(realmName);
+
+        if (currentRealm == null) {
+            currentRealm = realm;
         }
```

Comment:
We fallback to the realm inferred from the path if none/wrong was provided.

Reply:
I still feel that from the REST API perspective it would make more sense to explicitly tell that the realm doesn't exist rather than return something the user didn't ask for.

Reply:
But I don't want to block it, so let's leave it like it is. :)